Stack panics on invalid request body

<!--
Thanks for submitting a bug report. Please fill the template below,
otherwise we will not be able to process this bug report.
-->

#### Summary
<!-- Summarize the problem in a few sentences: -->

Stack panics on invalid request body when setting application collaborator.

#### Steps to Reproduce
<!-- How can we reproduce the problem? -->

Run stack with `tools/bin/mage dev:startDevStack`

Note that the body is missing the wrapping `collaborator` entry.
```
curl --location --request PUT 'http://localhost:1885/api/v3/applications/test-app/collaborators' \
--header 'Content-Type: application/json' \
--header 'Authorization: Bearer $access_token' \
--data-raw '{
    "ids": {
      "user_ids": {
        "user_id": "test-user3"
      }
    },
    "rights": [
      "RIGHT_APPLICATION_INFO",
    ]
}'
```

See the stack crashing with the following error:
```
panic: missed oneof type in extractCollaboratorFields()
```

<!--
Please upload relevant configuration (as .txt).
If you use the command "ttn-lw-stack config", you can redact sensitive config.
-->

#### What do you see now?
<!-- Please paste terminal output, upload logs (as .txt) or upload screenshots. -->

The stack is crashing

#### What do you want to see instead?
<!-- Please add some examples or mock-ups if applicable. -->

An error returning without panic

#### Environment
<!--
Your environment: OS/Browser/Gateway/Device/...? Versions? IDs/EUIs?
Paste the output of "ttn-lw-cli version" or "ttn-lw-stack version" if applicable.
-->

v3.11.3


#### How do you propose to test this?
<!-- Please think about how this is verified as fixed. -->

unit tests + manual

#### Can you do this yourself and submit a Pull Request?
<!-- You can also @mention experts if you need help with this. -->

@htdvisser or I can do if guided

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -49,6 +49,7 @@ For details about compatibility between different releases, see the **Commitment
 - Payload size limits for AU915 data rates 8 and 9, which are now consistent with Regional Parameters RP002-1.0.2.
 - Payload size limit calculation in Network Server.
 - Occasional panic in Network Server on downlink with corrupted device states.
+- Occasional panic in Identity Server on extracting log fields from invalid requests.
 
 ### Security
 

--- a/pkg/ttnpb/logfields.go
+++ b/pkg/ttnpb/logfields.go
@@ -62,6 +62,7 @@ func extractCollaboratorFields(m map[string]interface{}, ids *OrganizationOrUser
 		return
 	}
 	switch oneof := ids.Ids.(type) {
+	case nil:
 	case *OrganizationOrUserIdentifiers_OrganizationIDs:
 		m["collaborator_organization_id"] = oneof.OrganizationIDs.OrganizationID
 	case *OrganizationOrUserIdentifiers_UserIDs:

